### PR TITLE
Address selection from the KeyringController

### PR DESCRIPTION
**Pull request by gantunesr**
**Address selection from the KeyringController**

This PR updates the logic of the methods `addNewAccount`, `importAccountWithStrategy`, and `unlockQRHardwareWalletAccount` to return the last address added to the KeyringController. And also the method `forgetQRDevice` to return two arrays of addresses with the removed from the QR Keyring and remaining addresses.

**Description**

_Itemize the changes you have made into the categories below_

- BREAKING:

  - `KeyringController.addNewAccount`
    - Return new data type `Promise<{ addedAccountAddress: string; keyringState: KeyringMemState; }>`
  - `KeyringController.importAccountWithStrategy`
    - Return new data type `Promise<{ addedAccountAddress: string; keyringState: KeyringMemState; }>` 
  - `KeyringController.unlockQRHardwareWalletAccount`
    -  Return new data type `Promise<string>`
  - `KeyringController.forgetQRDevice`
    - Return new data type `Promise<{ removedAccounts: string[]; remainingAccounts: string[]; }>` 

**Checklist**

- [x] Tests are included if applicable
- [x] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: packages/keyring-controller/src/KeyringController.ts [hunk 1]
```diff
@@ -219,12 +222,14 @@ export class KeyringController extends BaseController<
     await this.verifySeedPhrase();
 
     this.updateIdentities(newAccounts);
+    let addedAccountAddress = '';
     newAccounts.forEach((selectedAddress: string) => {
       if (!oldAccounts.includes(selectedAddress)) {
-        this.setSelectedAddress(selectedAddress);
+        addedAccountAddress = selectedAddress;
```
Comment (mcmire) [inline packages/keyring-controller/src/KeyringController.ts:228]: This variable will change every time `!oldAccounts.includes(selectedAddress)`. Is that what we want?

In addition there don't seem to be any tests that check this logic or check that `addedAccountAddress` is non-empty, can those be added?

File: packages/keyring-controller/src/KeyringController.ts [hunk 2]
```diff
@@ -723,39 +732,49 @@ export class KeyringController extends BaseController<
...
     const newAccounts = await this.#keyring.getAccounts();
     this.updateIdentities(newAccounts);
+    let newHardwareWalletAddress = '';
     newAccounts.forEach((address: string) => {
       if (!oldAccounts.includes(address)) {
         if (this.setAccountLabel) {
           this.setAccountLabel(address, `${keyring.getName()} ${index}`);
         }
-        this.setSelectedAddress(address);
+        newHardwareWalletAddress = address;
```
Comment (mcmire) [inline packages/keyring-controller/src/KeyringController.ts:749]: Similar to a previous case, this variable will get updated every time `!oldAccounts.includes(address)`.

In addition it doesn't seem we are testing the return value of this method anywhere.

File: packages/keyring-controller/src/KeyringController.ts [hunk 3]
```diff
@@ -723,39 +732,49 @@ export class KeyringController extends BaseController<
...
-      this.setSelectedAddress(account);
-    });
+    const remainingAccounts = (await this.#keyring.getAccounts()) as string[];
+    const removedAccounts = accountsIncludingHardware.filter(
+      (address) => !remainingAccounts.includes(address),
+    );
+    this.updateIdentities(remainingAccounts);
     await this.#keyring.persistAllKeyrings();
     await this.fullUpdate();
+    // Return list of removed accounts as well as remaining accounts for the app to use.
```
Comment (mcmire) [inline packages/keyring-controller/src/KeyringController.ts:776]: This comment doesn't seem to add any more information than the variables themselves; thoughts on removing?

**Review by mcmire — COMMENTED** [review]
Seems straightforward — just had a few comments.

File: packages/keyring-controller/src/KeyringController.ts (hunk 3 above)
Comment (gantunesr) [reply packages/keyring-controller/src/KeyringController.ts]: Agreed. Removed on [40dc630](https://github.com/MetaMask/core/pull/1095/commits/40dc630149c459e309a1f5368b5ee707cf59404f)

File: packages/keyring-controller/src/KeyringController.ts (hunk 1 above)
Comment (gantunesr) [reply packages/keyring-controller/src/KeyringController.ts]: What do you think about this refactor? [73c539d](https://github.com/MetaMask/core/pull/1095/commits/73c539d9f8b3bdabdf44054e275acb36b71d0c10)

I'll await until I get your input before developing the proper test

File: packages/keyring-controller/src/KeyringController.ts (hunk 2 above)
Comment (gantunesr) [reply packages/keyring-controller/src/KeyringController.ts]: Same here.

What do you think about this refactor? [b65201a](https://github.com/MetaMask/core/pull/1095/commits/b65201a505b537138003f24ba2497d6f8f1bf84d)

I'll await until I get your input before developing the proper test

File: packages/keyring-controller/src/KeyringController.ts (hunk 1 above)
Comment (mcmire) [reply packages/keyring-controller/src/KeyringController.ts]: Yeah that refactor makes more sense!

File: packages/keyring-controller/src/KeyringController.ts (hunk 2 above)
Comment (mcmire) [reply packages/keyring-controller/src/KeyringController.ts]: Yeah that makes a lot more sense.

**Comment by gantunesr** [comment]
This PR is no longer relevant

**Pull request closed without merging**